packages/driver/src/cy/commands/querying/querying.ts

```diff
@@ -142,13 +144,23 @@ function getAlias (selector, log, cy) {
 }
 
 export default (Commands, Cypress, cy, state) => {
+  function validateTimeoutFromOpts (options: GetOptions | ContainsOptions | ShadowOptions = {}, queryCommand: QueryCommandOptions = '') {
+    if (!isEmpty(queryCommand) && _.isPlainObject(options) && options.hasOwnProperty('timeout') && !_.isFinite(options.timeout)) {
+      $errUtils.throwErrByPath(`${queryCommand}.invalid_option_timeout`, {
+        args: { timeout: options.timeout },
+      })
+    }
+  }
+
```

Comment:
Since it does not reference any values via closure, it's possible to pull this function up to the top level so it does not get re-created each time the default export is called.

```suggestion

function validateTimeoutFromOpts (options: GetOptions | ContainsOptions | ShadowOptions = {}, queryCommand: QueryCommandOptions = '') {
  if (!_.isEmpty(queryCommand) && _.isPlainObject(options) && options.hasOwnProperty('timeout') && !_.isFinite(options.timeout)) {
    $errUtils.throwErrByPath(`${queryCommand}.invalid_option_timeout`, {
      args: { timeout: options.timeout },
    })
  }
}

export default (Commands, Cypress, cy, state) => {
```

Reply:
Oh yeah, makes sense.

Reply:
This is addressed, thanks for the review.

Reply:
Thank you!